sklearn/model_selection/tests/test_search.py

```diff
@@ -2079,3 +2085,56 @@ def _fit_param_callable():
         'scalar_param': 42,
     }
     model.fit(X_train, y_train, **fit_params)
+
+
+def _generate_search_cv_using_minimal_compatible_instances():
+    """Generate instance containing estimators from minimal class compatible
+    implementation that should be supported by `SearhCV`."""
+    for SearchCV, (Estimator, param_grid) in zip(
+        [GridSearchCV, RandomizedSearchCV],
+        [
+            (MinimalRegressor, {"param": [1, 10]}),
+            (MinimalClassifier, {"param": [1, 10]}),
+        ],
+    ):
+        yield SearchCV(Estimator(), param_grid)
+
+    for SearchCV, (Estimator, param_grid) in zip(
+        [GridSearchCV, RandomizedSearchCV],
+        [
+            (
+                MinimalRegressor,
+                {
+                    "minimaltransformer__param": [1, 10],
+                    "minimalregressor__param": [1, 10],
+                },
+            ),
+            (
+                MinimalClassifier,
+                {
+                    "minimaltransformer__param": [1, 10],
+                    "minimalclassifier__param": [1, 10],
+                },
+            ),
+        ],
+    ):
+        yield SearchCV(
+            make_pipeline(MinimalTransformer(), Estimator()), param_grid
+        ).set_params(error_score="raise")
+
+
+# FIXME: hopefully in 0.25
+@pytest.mark.xfail(
+    reason=(
+        "This test is currently failing because checks are granular "
+        "enough. Once checks are split with some kind of only API tests, "
+        "this test should enabled."
+    )
+)
+@parametrize_with_checks(
+    list(_generate_search_cv_using_minimal_compatible_instances())
+)
+def test_search_cv_using_minimal_compatible_estimator(estimator, check):
+    # Check that third-party library can run tests without inheriting from
+    # BaseEstimator.
+    check(estimator)
```

Comment:
I am not opposed to run the full `parametrize_with_checks` check-suite but before we get to refactor the check estimator tool to support the API compliance mode, wouldn't it be possible to just check that it's possible to use those minimal estimators in a grid search by calling fit without crashing.

What I mean is that we could just writing a few manual smoke tests that do not depend on the estimator checks machinery:

- one pipeline with MinimalTransformer followed by MinimalClassifier
- one pipeline with MinimalTransformer followed by MinimalRegressor

For each of those, we wrap them in a `GridSearchCV` on a toy dataset and we call fit / predict / score and make a basic assertion on the content of `best_params_`.